src/main/java/com/ibm/cloud/sdk/core/http/gzip/GzipRequestInterceptor.java

```diff
@@ -0,0 +1,45 @@
+package com.ibm.cloud.sdk.core.http.gzip;
+
+import java.io.IOException;
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okio.BufferedSink;
+import okio.GzipSink;
+import okio.Okio;
+
+/** This interceptor compresses the HTTP request body. Many webservers can't handle this! */
```

Comment:
Should probably add a reference to where this class was copied from:
https://square.github.io/okhttp/interceptors/#rewriting-requests